### PR TITLE
Docs/simplifying docs

### PR DESCRIPTION
**Pull request by Adeniyikayodee**
# Reason for This Change

Elaborate on the reasons for this change. If applicable link it
to the related issue (like `Closes #<issue>`).

Note: We are not accepting feature PR's for the time being (this will
change in the future).

## Description of Changes

Overview of what was changed.

### CODE REVIEW

File: docs/stacks/index.md [hunk 1]
```diff
@@ -13,30 +13,82 @@ next:
...
+
+- To **update** the stack, use:
+```hcl
+terramate stack update
+```
+
+- To **delete** the entire stack you can use the command:
+```hcl
+terramate stack delete
+```
```
Comment (soerenmartius) [inline docs/stacks/index.md:79]: `terramate stack` does not exist actually. I am wondering where this is coming from and how you came with this? So far we have `terramate create` for creating new stacks.

File: docs/stacks/index.md [hunk 2]
```diff
@@ -13,30 +13,82 @@ next:
...
+## Creating a Stack:
+
+To create a stack using Terramate, follow these steps:
+
+1. Define the included resources in a Terraform configuration file. This file should contain all resources you want to manage as part of the stack.
+
+2. Use the following command to create the resources:
+
+```hcl
+terramate stack create
```
Comment (soerenmartius) [inline docs/stacks/index.md:63]:
```suggestion
terramate create
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

**Comment by soerenmartius** [comment]
@Adeniyikayodee thanks for the contribution. Regarding your commits messages, we're following https://www.conventionalcommits.org/en/v1.0.0/ . `edit:` doesn't exist in the conventialcommit standard, but you can use `docs:` instead.

File: docs/about-stacks.md [hunk 3]
```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
 the infrastructure's resources.
 ```
 
-By _runnable_ it means it has everything needed to call
-`terraform init/plan/apply` . For example, a module used by other modules, ie,
-don't have the `provider` explicitly set, is not runnable hence it's
-**not a stack**.
+For a module to be considered a stack, it must be complete and self-contained, 
+meaning that it can run all the necessary Terraform commands (`init/plan/apply`) 
+without any external dependencies.
```
Comment (soerenmartius) [inline docs/about-stacks.md:29]: Shall we make more clear that provider and Terraform configuration is required? I would make this way more explicit, especially to distinguish between modules and stacks. wdyt @mariux ?

File: docs/about-stacks.md [hunk 4]
```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
 
-If your runnable Terraform module creates your whole infrastructure, *it's
-also not a stack*, since the idea of stacks is to be a unit of infrastructure
-decomposition.
+If a module is missing any required components, such as provider settings,
```
Comment (soerenmartius) [inline docs/about-stacks.md:31]: Here I would also say that we explicitly list mandatory components such as provider settings and optional components such as state

File: docs/about-stacks.md [hunk 5]
```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
+without any external dependencies.
 
-If your runnable Terraform module creates your whole infrastructure, *it's
-also not a stack*, since the idea of stacks is to be a unit of infrastructure
-decomposition.
+If a module is missing any required components, such as provider settings, 
+it cannot be considered a stack.
 
-## Isolate code with higher change frequency
+It's also worth noting that if a module creates your entire infrastructure,
```
Comment (soerenmartius) [inline docs/about-stacks.md:34]: nice!

File: docs/about-stacks.md (hunk 3 above)
Comment (mariux) [reply docs/about-stacks.md:29]: A stack is not limited to terrform only.. it can be a kubernetes stack, pure Terraform, Terrgraunt or anything..

In Terraform context (most common) a stack is what hashicorop call a root module.. it has a single state by default but when using workspaces can have multiple states (one per workspace)..

Terraform defaults to a local backen if none is defined.. Terraform also just downloads the latest version of all used providers it is aware of, so most configurations work without defining any of those.. 

Best practice demand to add such cibfiguratio anyway and terraform will warn if it is not defined but not fail.. 

Terramate integrates with Terraform stacks the beat atm as change detection is aware of any dependendciesike local modules.. 

Terraform itself does not fully distinguish between stacks (root modules) and modules itself.. with the newly released zero config modules it actually breaks with everything that was before a good practice.. ;)

Tereamte does not demand anything but having a stack config block defined in any Terramate config file within the stack.. 

using tags you can define intelligent filtering to even run multiple types stacks in a single repository.. or having a single stack being a kubernetes and Terraform stack at the same time.. 

We can define best practices though for each stack type.. e.g. a Terramate Terraform Stack should have provider and backend config defined and include versioning constraints for providers and Terraform itself.. 

Bottom line is, we are not limited to Terraform.. and do not demand anything but an (Empty) stack block as a minimal setup.. at least in Terramte CLI this is enough to get it going.. ;) 

We will add better onbaording in a soon to come Terramte release for existing Terraform configurations.. Also provide easy ways to add a inquiry stack id to existing stacks that miss one..

File: docs/about-stacks.md [hunk 6]
```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
-apply the change in your single Terraform project, some dependent resources can
-be recreated leading to increased downtime. There are several reasons why a
-dependent module can be recreated, eg.: attributes with variable interpolation
-from recreated resources; underspecified attributes refreshing during plan, etc.
+## Isolate code that changes frequently
 
-By using stacks you can modify only the stacks affected by the deploys or
-configuration changes needed, but you have to choose the size of the stack
-wisely to avoid duplication.
+If you make a lot of updates to your system, like deploying new code or changing
```
Comment (i4ki) [inline docs/about-stacks.md:41]: *your system* is way off here, no? 
*System* in computing is often used when referring to software that's part of the operating system.

File: docs/about-stacks.md [hunk 7]
```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
-dependent module can be recreated, eg.: attributes with variable interpolation
-from recreated resources; underspecified attributes refreshing during plan, etc.
+## Isolate code that changes frequently
 
-By using stacks you can modify only the stacks affected by the deploys or
-configuration changes needed, but you have to choose the size of the stack
-wisely to avoid duplication.
+If you make a lot of updates to your system, like deploying new code or changing 
+settings often, doing everything in one Terraform project could cause problems. 
+This might lead to some parts of your system needing to be rebuilt, which means
```
Comment (i4ki) [inline docs/about-stacks.md:43]: here again, the wording imply Terramate has something to do with applying settings in the server or in the user's machine.

**Comment by Adeniyikayodee** [comment]
All reviews except mandatory and optional configurations have been resolved. Further explanation for stacks will be included in the coming merge request.

**Comment by soerenmartius** [comment]
> All reviews except mandatory and optional configurations have been resolved. Further explanation for stacks will be included in the coming merge request.

amazing thank you

**Comment by Adeniyikayodee** [comment]
@soerenmartius, please approve this pull request.

**Review by soerenmartius — APPROVED** [review]
lgtm

File: docs/stacks/index.md [hunk 8]
```diff
@@ -13,30 +13,62 @@ next:
...
+
+2. Use the following command to create the resources:
+
+```hcl
+terramate create
+```
+Running this command creates all defined resources.
+
+
+# Properties
```
Comment (soerenmartius) [inline docs/stacks/index.md:68]:
```suggestion
# Properties

```

**Comment by soerenmartius** [comment]
> @soerenmartius, please approve this pull request.

done :)

File: docs/about-stacks.md [hunk 9]
```diff
@@ -24,47 +24,48 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
+settings often, doing everything in one Terraform project could cause problems. 
+This might lead to some parts of your infrastructure needing to be rebuilt, which means 
+more downtime.
 
-The overall principle being, code that changes at different rate change for
-different reasons and code that changes for different reasons should be as
-isolated from each other as possible.
+If you frequently update your infrastructure, such as deploying new code or modifying settings, 
+doing everything into one Terraform project could cause complications. This might lead to 
+some parts of your infrastructure needing to be rebuilt, which means increased downtime.
```
Comment (soerenmartius) [inline docs/about-stacks.md:48]: @Adeniyikayodee 

This seems to be duplicated?

File: docs/getting-started/index.md [hunk 10]
```diff
@@ -106,25 +111,29 @@ globals {
 }
 ```
 
-Note that there is nothing special about the name `globals.tm.hcl` and we could have put it in the existing `terramate.tm.hcl` because all Terramate files are merged at runtime (similar to how Terraform merges `.tf` files). This simplicity allows great flexibility to fit the way you choose to work.
+The name `globals.tm.hcl` is not special, and the content could be included in an existing `terramate.tm.hcl` file. All Terramate files are merged during runtime, similar to how Terraform merges `.tf` files. This simplicity provides great flexibility.
```
Comment (soerenmartius) [inline docs/getting-started/index.md:114]:
```suggestion
The name `globals.tm.hcl` is not required, and the content could be included in an existing `terramate.tm.hcl` file. All Terramate files are merged during runtime, similar to how Terraform merges `.tf` files. This simplicity provides great flexibility.
```

Maybe required makes more sense than special? Just an idea

File: docs/getting-started/index.md [hunk 11]
```diff
@@ -204,38 +213,42 @@ generate_hcl "mysite.tf" {
...
 $ terramate run terraform apply
 ```
 
-After approving, you should now have the rendered HTML files at `/tmp/tfmysite/<env>/mysite/index.html`
+After approval, the rendered HTML files should be located at `/tmp/tfmysite/<env>/mysite/index.html`
 
 ### Terramate Orchestration with Change Detection
 
-To finish this quick introduction, let’s look at the globals precedence and one of Terramate’s most powerful features: change detection. Change detection is expected to be run in a CI-CD pipeline and requires a working git with a remote repository, so go to the root of your project and run
+To conclude this introduction, let's explore globals precedence and one of Terramate's most powerful features: change detection. Change detection is designed for CI-CD pipelines and requires a working Git repository with a remote. In your project's root, run:
```
Comment (soerenmartius) [inline docs/getting-started/index.md:243]:
```suggestion
To conclude this introduction, let's explore globals precedence and one of Terramate's most powerful features: change detection. [Change detection](../change-detection/index.md) is designed for CI-CD pipelines and requires a working Git repository with a remote. In your project's root, run:
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: docs/about-stacks.md (hunk 9 above)
Comment (Adeniyikayodee) [reply docs/about-stacks.md:48]: Yes, that was a duplicate. I have removed it on my end, it will come with the next PR. Thank you for painstakingly reviewing the docs.

File: docs/about-stacks.md [hunk 12]
```diff
@@ -24,47 +24,48 @@ A terraform stack is a runnable terraform module that operates on a subset of
...
 An extreme example is: avoiding a database instance destroy because of a DNS TTL
 change.
 
 ## Reduce execution time
 
-By using stacks you can reduce the time an infrastructure change takes to finish.
-This is even more appealing if your Terraform changes are applied through CI
-builders running in the cloud because faster integration/builds leads to reduced
-cloud costs.
+Using stacks helps speed up infrastructure changes. This is especially beneficial when applying Terraform changes through cloud-based CI builders, as quicker integration and builds can lower cloud costs.
```
Comment (soerenmartius) [inline docs/about-stacks.md:68]:
```suggestion
Using stacks allows you to reduce the runtime of Terraform drastically. This is especially beneficial when running Terraform in CI/CD environments like GitHub Actions since build time is usually billed per minute, which means faster execution time translates directly to lowered costs.
```

I would write it like this, way more explicit and understandable

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: docs/getting-started/index.md [hunk 13]
```diff
@@ -15,18 +15,21 @@ next:
...
 >
 > 👉 [https://terramate.io/discord](https://terramate.io/discord)
 
 ## Set up Terramate
 
-> If you haven’t installed Terramate yet, please chose one of the various existing ways of how to install Terramate in ["Installation"](../installation.md).
+> If you haven’t installed Terramate yet, please refer to the [installation](../installation.md) section for various  options. 
 
-Start by creating a new directory and create a file there called `terramate.tm.hcl` with the contents:
+To begin, create a new directory, and within it, create a file named `terramate.tm.hcl` with the following content:
```
Comment (soerenmartius) [inline docs/getting-started/index.md:32]:
```suggestion
To begin, create a new directory that contains a file named `terramate.tm.hcl` with the following content:
```
Just an idea.. not mandatory to apply this. Reads easier from my perspective

File: docs/getting-started/index.md [hunk 14]
```diff
@@ -58,17 +59,13 @@ stack {
...
-
-A Terramate Stack is a directory that contains Terramate configuration files. This is where Terramate will generate Terraform. A stack has a 1:1 relationship with the Terraform state - i.e. each stack is a directory where you can run `terraform init`
-
-Moreover, there is nothing special about this `stack.tm.hcl` file and it can be generated manually without the ID or the `terramate create` command.
+Terramate recognizes files with the `.tm` or `.tm.hcl` extension, and any file containing a `stack {}` block is considered a stack. A stack is simply a directory where Terramate generates Terraform files. 
 
-However, it’s preferable to generate unique IDs for our stacks because, as we build more complex infrastructures, they allow us to reference stacks directly rather than using relative paths, which makes refactoring painful.
+You can create the `stack.tm.hcl` file manually without an ID or by running the `terramate create` command, but generating unique IDs for stacks is recommended. Unique IDs enable direct stack referencing instead of using relative paths, which can make refactoring difficult.
```
Comment (soerenmartius) [inline docs/getting-started/index.md:64]:
```suggestion
> **Note:** You can manually create the `stack.tm.hcl` file without an ID or by running the `terramate create` command.
> However, generating unique IDs for stacks is recommended since they enable direct stack referencing instead of relative paths, making refactoring difficult.
```

I would add this as a quote or so and make sure that the user understands it's optional but not the desired way

File: docs/orchestration/index.md [hunk 15]
```diff
@@ -233,103 +169,26 @@ stack {
...
-execution is **terramate run**.
-
-This will run **terraform** plan on all stacks, but respecting ordering:
+One command that considers this execution order is `terramate run`, which runs `terraform plan` on all stacks in the defined sequence:
 
 ```sh
 terramate run terraform plan
 ```
 
-### Change Detection And Ordering
```
Comment (soerenmartius) [inline docs/orchestration/index.md:251]: hmm why do we delete all of this? Please explain :)

File: docs/stacks/index.md [hunk 16]
```diff
@@ -13,30 +13,63 @@ next:
...
-It doesn't require any attributes by default, but it can be used
-to describe stacks and orchestrate their execution.
+## What is a Stack?
+
+A stack is a collection of related resources managed as a single unit. When defining a stack, you specify all included resources, such as networks, virtual machines, and storage. 
+
+Terramate CLI simplifies resource creation and management, allowing you to focus on building and deploying the application.
+
+
+## A Terramate stack is:
```
Comment (soerenmartius) [inline docs/stacks/index.md:30]:
```suggestion
## A Terramate Stack is:
```

File: docs/getting-started/index.md (hunk 14 above)
Comment (Adeniyikayodee) [reply docs/getting-started/index.md:64]: Yes, this makes sense. Get readers' attention and conveys the message easily.

File: docs/orchestration/index.md (hunk 15 above)
Comment (Adeniyikayodee) [reply docs/orchestration/index.md:251]: I chose not to discuss this topic in the Orchestration Section because it will be fully discussed in the next section, which is about Change Detection. I did this to follow best practices. The aim was to concentrate on one idea and give a detailed guide on how to apply it on each page, instead of mixing up many ideas in one place.

**Review by soerenmartius — APPROVED** [review]
lgtm - good job - let's get this merged

File: docs/about-stacks.md (hunk 9 above)
Comment (i4ki) [inline docs/about-stacks.md:48]: this phrase is repeated with different wording..

**Review by i4ki — CHANGES_REQUESTED** [review]

File: docs/orchestration/index.md [hunk 17]
```diff
@@ -56,68 +51,37 @@ applied is: `change detection`, `current directory`, `wants`.
...
-the fields **before** and **after**.
-
-Each field is a set of string (**set(string)**), where each string is a path that
-references another directory, which can be a stack or contain stacks inside.
-
-The explicit order can be used in conjunction with the implicit filesystem order
-but have in mind that a parent stack in the filesystem can never execute after a
-child one, and trying to make this using explicit **before** and **after** clauses
-will lead to a cycle error.
+Terramate's Explicit Order of Execution feature allows you to designate a specific order for stack execution. This is done by definin **before** and **after** fields within the **stack** block, referring to other directories containing stacks.
```
Comment (i4ki) [inline docs/orchestration/index.md:72]:
```suggestion
Terramate's Explicit Order of Execution feature allows you to designate a specific order for stack execution. This is done by defining **before** and **after** fields within the **stack** block, referring to other directories containing stacks.
```

File: docs/orchestration/index.md [hunk 18]
```diff
@@ -56,68 +51,37 @@ applied is: `change detection`, `current directory`, `wants`.
...
-"I execute before all stacks inside these directories".
-
-**after** ensures the opposite, that the stacks contained in the provided
-directories are executed before the current stack. As the stack, you are saying:
-"I execute after all stacks inside these directories".
+1. Open your Terramate configuration file `terramate.tm.hcl` in a text editor.
+2. Select the stack block you want to allocate a specific order.
+3. Depending on your needs, add either the **before** or **after** field inside the stack block.
+4. Add **string** values representing the directory paths you want to order execution for in the **before** or **after** field.
+5. Save and close the file.
```
Comment (i4ki) [inline docs/orchestration/index.md:84]: those steps don't make much sense.
Step 1 and 2 are wrong. You have to choose the stack directory you want to change the execution order, and then open the file that contains the stack definition (usually `stack.tm.hcl`)

This text comes before the example below that shows that stacks are defined in the *terramate.tm.hcl* files.

File: docs/about-stacks.md (hunk 9 above)
Comment (soerenmartius) [reply docs/about-stacks.md:48]: > this phrase is repeated with different wording..

@i4ki what's your suggestion in that case? Shall we revert this to the previous version or would you like to suggest a more clear wording?

File: docs/orchestration/index.md [hunk 19]
```diff
@@ -233,103 +169,26 @@ stack {
...
 
-* stack-a
-* stack-c
-* stack-b
-
-One example of terramate command that leverages order of
-execution is **terramate run**.
-
-This will run **terraform** plan on all stacks, but respecting ordering:
+One command that considers this execution order is `terramate run`, which runs `terraform plan` on all stacks in the defined sequence:
```
Comment (i4ki) [inline docs/orchestration/index.md:174]: this phrase is misleading as it imply that `terramate run` automatically runs `terraform plan` on all stacks..

File: docs/orchestration/index.md [hunk 20]
```diff
@@ -233,103 +169,26 @@ stack {
...
-Currently the behavior when a stack execution fails given a total order of
-execution is undefined. Whatever observed behavior should not be relied upon
-since it may change on the future.
+The current behavior during complete failure in stack order execution remains undefined. 
+As this behavior is subject to change, it is advisable not to rely on it.
 
+### Handling Cycles Conflicts
 
-### What About Cycles/Conflicts ?
+Cyclic dependency occurs when circular dependencies exist between the stacks. For instance, if stack A is dependent on stack B, and vice versa, a cycle is created.
```
Comment (i4ki) [inline docs/orchestration/index.md:187]: > Cyclic dependency occurs when circular dependencies exist between the stacks.

I think this should be removed or improved, as it's also a cyclic definition :)

File: docs/orchestration/index.md [hunk 21]
```diff
@@ -342,4 +201,4 @@ stack {
 }
 ```
 
-An error will be reported.
+In this situation, a conflict arises causing the execution to enter failure mode. An error message will be reported in such an instance.
```
Comment (i4ki) [inline docs/orchestration/index.md:204]: Not specifically about your change here, as the old text wasn't ideal also, but it'd be better to explain that it's a fatal error, not just an error message.

File: docs/stacks/index.md [hunk 22]
```diff
@@ -13,30 +13,63 @@ next:
...
+Terramate CLI simplifies resource creation and management, allowing you to focus on building and deploying the application.
+
+
+## A Terramate Stack is:
+
+- A directory inside your project
+- Contains one or more Terramate configuration files
+- Includes a configuration file with a stack{} block
+
+The `stack{}` block distinguishes a stack from other directories in Terramate. By default, it doesn't require any attributes but can be used to describe stacks and orchestrate their execution.
```
Comment (i4ki) [inline docs/stacks/index.md:36]:
```suggestion
The `stack{}` block distinguishes a stack from other directories in Terramate. By default, it doesn't require any attributes but can be used to describe the stack and orchestrate its execution.
```
The `stack` block defines a single stack.

File: docs/stacks/index.md [hunk 23]
```diff
@@ -13,30 +13,63 @@ next:
...
+The `stack{}` block also defines attributes used to describe the stack. Only [Terramate Functions](../functions/index.md) are available when defining
 the `stack` block.
 
+## Why use Stacks?
+
+Stacks offer several benefits:
+
+1. Manage multiple resources as one unit, allowing you to build and deploy the entire infrastructure efficiently and quickly with a single command.
+
+2. Easily manage dependencies between resources. For example, you can specify a virtual machine that depends on a virtual network, and Terramate CLI will automatically manage the resource's creation or deletion.
```
Comment (i4ki) [inline docs/stacks/index.md:49]: this part `Terramate CLI will automatically manage the resource's creation or deletion.` is not entirely correct... This could be misleading for new users because they could think that Terramate does some kind of automatic resource management.

File: docs/about-stacks.md (hunk 9 above)
Comment (i4ki) [reply docs/about-stacks.md:48]: I meant that it's duplicated, sorry

Comment (i4ki) [reply docs/about-stacks.md:48]:
```
If you make a lot of updates to your infrastructure, like deploying new code or changing settings often, doing everything in one Terraform project could cause problems. This might lead to some parts of your infrastructure needing to be rebuilt, which means more downtime.

If you frequently update your infrastructure, such as deploying new code or modifying settings, doing everything into one Terraform project could cause complications. This might lead to some parts of your infrastructure needing to be rebuilt, which means increased downtime.
```

See here: https://github.com/mineiros-io/terramate/blob/b9345b6258b8ac37d956a03505adb1a610a363eb/docs/about-stacks.md#isolate-code-that-changes-frequently

Comment (soerenmartius) [reply docs/about-stacks.md:48]: that makes sense - thank you

**Comment by soerenmartius** [comment]
Closing this as the scope of the PR got out of hand slightly. Will add the useful contributions in smaller scoped PRs
